### ISSUE TITLE
TypeScript parsing error: useState<Engagement[]>()

### ISSUE THREAD

**Issue by nekinie** [semgrep/semgrep#3006]
**Describe the bug**
Semgrep fails to parse the typescript line:
```
const [engagements, setEngagements] = useState<Engagement[]>()
```
Resulting in the output:
```
using config from https://semgrep.dev/p/r2c-ci. Visit https://semgrep.dev/registry to see all public rules.
downloading config...
using config from https://semgrep.dev/p/react. Visit https://semgrep.dev/registry to see all public rules.
downloading config...
excluding files:
- tools.go
running 152 rules...
semgrep warn: parse error
  --> web/src/components/ReportVulnerability.tsx:51
51 |   const [engagements, setEngagements] = useState<Engagement[]>()
   |                                                               ^^^^^^^^^
= help: If the code appears to be valid, this may be a semgrep bug.
Could not parse ReportVulnerability.tsx as ts

semgrep warn: parse error
  --> web/src/components/ReportVulnerability.tsx:51
51 |   const [engagements, setEngagements] = useState<Engagement[]>()
   |                                                               ^^^^^^^^^
= help: If the code appears to be valid, this may be a semgrep bug.
Could not parse ReportVulnerability.tsx as typescript

ran 152 rules on 85 files: 0 findings
```

**Expected behavior**
No parsing error.

**What is the priority of the bug to you?**
P1

**Environment**
Docker returntocorp/semgrep:0.48.0

**Comment by mjambon** [comment]
This code parses without an error with the latest grammar for TSX, which should be integrated into semgrep sometime in the coming weeks.

```
~/tree-sitter-typescript/tsx $ ../node_modules/.bin/tree-sitter parse <(echo 'const [engagements, setEngagements] = useState<Engagement[]>()')
(program [0, 0] - [1, 0]
  (lexical_declaration [0, 0] - [0, 62]
    (variable_declarator [0, 6] - [0, 62]
      name: (array_pattern [0, 6] - [0, 35]
        (identifier [0, 7] - [0, 18])
        (identifier [0, 20] - [0, 34]))
      value: (call_expression [0, 38] - [0, 62]
        function: (identifier [0, 38] - [0, 46])
        type_arguments: (type_arguments [0, 46] - [0, 60]
          (array_type [0, 47] - [0, 59]
            (type_identifier [0, 47] - [0, 57])))
        arguments: (arguments [0, 60] - [0, 62])))))
```

**Comment by mschwager** [comment]
Related: https://github.com/returntocorp/semgrep/issues/3033

**Comment by mjambon** [comment]
The fix will be in semgrep 0.56 due this week.
